api/v1alpha1/networkvalidator_types.go

```diff
@@ -42,15 +42,49 @@ type NetworkValidatorSpec struct {
 	// +kubebuilder:validation:MaxItems=5
 	// +kubebuilder:validation:XValidation:message="TCPConnRules must have unique names",rule="self.all(e, size(self.filter(x, x.name == e.name)) == 1)"
 	TCPConnRules []TCPConnRule `json:"tcpConnRules,omitempty" yaml:"tcpConnRules,omitempty"`
+	// HTTPFileRules validate that files are publicly available via HTTP
+	// +kubebuilder:validation:MaxItems=5
+	// +kubebuilder:validation:XValidation:message="HTTPFileRules must have unique names",rule="self.all(e, size(self.filter(x, x.name == e.name)) == 1)"
+	HTTPFileRules []HTTPFileRule `json:"httpFileRules,omitempty" yaml:"httpFileRules,omitempty"`
+	// CACerts allow additional CA certificates to be used for TLS. Applies to TCPConnRules and HTTPFileRules.
+	CACerts CACertificates `json:"caCerts,omitempty" yaml:"caCerts,omitempty"`
+}
+
+// CACertificates contains configuration for additional CA certificates to use for TLS. Can be certs
+// provided inline or secret references. Secrets are assumed to be in the same namespace as the
+// NetworkValidator.
+type CACertificates struct {
+	// Certs is a list of certificates to use.
+	// +kubebuilder:validation:MaxItems=500
+	Certs []Certificate `json:"certs,omitempty" yaml:"certs,omitempty"`
+	// SecretRefs is a list of secret references to use.
+	// +kubebuilder:validation:MaxItems=500
+	SecretRefs []SecretReference `json:"secretRefs,omitempty" yaml:"secretRefs,omitempty"`
+}
+
+// Certificate is a certificate specified inline.
+// +kubebuilder:validation:MinLength=1
+type Certificate string
```

Comment:
I don't think this type is adding any value.

Reply:
It's for `+kubebuilder:validation:MinLength=1`. If you have a slice of strings, you can't use that validation rule to control min length of each string. This is the workaround they've been recommending.

If we don't need this validation, I can take it out though.